psydac/linalg/solvers.py

```diff
@@ -127,16 +132,30 @@ class ConjugateGradient(InverseLinearOperator):
     [1] A. Maister, Numerik linearer Gleichungssysteme, Springer ed. 2015.
 
     """
-    def __init__(self, A, *, x0=None, tol=1e-6, maxiter=1000, verbose=False, recycle=False):
+    def __init__(self, A, *, pc=None, x0=None, tol=1e-6, maxiter=1000, verbose=False, recycle=False):
 
-        self._options = {"x0":x0, "tol":tol, "maxiter":maxiter, "verbose":verbose, "recycle":recycle}
+        self._options = {"x0":x0, "pc": pc, "tol":tol, "maxiter":maxiter, "verbose":verbose, "recycle":recycle}
         
         super().__init__(A, **self._options)
-        
-        self._tmps = {key: self.domain.zeros() for key in ("v", "r", "p")}
+
+        if pc is None: 
+            self._tmps = {key: self.domain.zeros() for key in ("v", "r", "p")}
+
+        else: 
+            assert isinstance(pc, LinearOperator)
```

Comment:
We should check that domain and codomain of this linear operator are correct:

```suggestion
            assert isinstance(pc, LinearOperator)
            assert pc.domain is A.codomain
            assert pc.codomain is A.domain
```